src/jllvm/vm/StringInterner.hpp

```diff
@@ -0,0 +1,34 @@
+#pragma once
+
+#include <llvm/Support/raw_ostream.h>
+
+#include <jllvm/object/ClassLoader.hpp>
+#include <jllvm/support/Encoding.hpp>
+
+#include <map>
+
+namespace jllvm
+{
+class StringInterner
+{
+    static constexpr auto stringDescriptor = "Ljava/lang/String;";
+    static constexpr auto byteArrayDescriptor = "[B";
+
+    std::map<llvm::StringRef, String*> m_literalToStringMap;
```

Comment:
Could you leave a TODO above here that this has to changed to a `(encoding, bytes)` pair as key for proper interning?
` std::map<llvm::StringRef,..>` is also a bit risky lifetime wise since there is no guarantee from a callers perspective that the `StringRef`s storage stays alive, but should be fine at the moment since the `ClassFile`s `StringRef`s are explicitly kept alive

Reply:
It has been changed